src/main/java/net/mcreator/ui/modgui/ItemExtensionGUI.java

```diff
@@ -0,0 +1,168 @@
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.ui.modgui;
+
+import net.mcreator.blockly.data.Dependency;
+import net.mcreator.element.types.ItemExtension;
+import net.mcreator.minecraft.ElementUtil;
+import net.mcreator.ui.MCreator;
+import net.mcreator.ui.MCreatorApplication;
+import net.mcreator.ui.component.util.PanelUtils;
+import net.mcreator.ui.help.HelpUtils;
+import net.mcreator.ui.init.L10N;
+import net.mcreator.ui.minecraft.MCItemHolder;
+import net.mcreator.ui.procedure.ProcedureSelector;
+import net.mcreator.ui.validation.AggregatedValidationResult;
+import net.mcreator.workspace.elements.ModElement;
+import net.mcreator.workspace.elements.VariableTypeLoader;
+
+import javax.annotation.Nullable;
+import javax.swing.*;
+import javax.swing.border.TitledBorder;
+import java.awt.*;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+public class ItemExtensionGUI extends ModElementGUI<ItemExtension> {
+	private final MCItemHolder item = new MCItemHolder(mcreator, ElementUtil::loadBlocksAndItems);
+
+	// Fuel
+	private final JCheckBox enableFuel = L10N.checkbox("elementgui.common.enable");
+	private final JSpinner fuelPower = new JSpinner(new SpinnerNumberModel(1600, 0, Integer.MAX_VALUE, 1));
+
+	// Composter
+	private final JSpinner layerChance = new JSpinner(new SpinnerNumberModel(0, 0.00f, 1.00f, 0.05f));
+
+	// Dispenser behaviour
+	private final JCheckBox hasDispenseBehavior = L10N.checkbox("elementgui.common.enable");
+	private ProcedureSelector dispenseSuccessCondition;
+	private ProcedureSelector dispenseResultItemstack;
+
+	public ItemExtensionGUI(MCreator mcreator, ModElement modElement, boolean editingMode) {
+		super(mcreator, modElement, editingMode);
+		this.initGUI();
+		super.finalizeGUI();
+	}
+
+	@Override protected void initGUI() {
+		//Fuel
+		JPanel fuelPanel = new JPanel(new BorderLayout());
+		fuelPanel.setOpaque(false);
+
+		JComponent enableFuelComp = PanelUtils.gridElements(1, 2, 0, 5,
+				HelpUtils.wrapWithHelpButton(this.withEntry("item_extension/enable_fuel"),
+						L10N.label("elementgui.item_extension.enable_fuel")), enableFuel);
+		enableFuel.setOpaque(false);
+		enableFuel.addActionListener(e -> fuelPower.setEnabled(enableFuel.isSelected()));
+		enableFuelComp.setOpaque(false);
+		fuelPanel.add(PanelUtils.totalCenterInPanel(PanelUtils.northAndCenterElement(enableFuelComp,
+				PanelUtils.westAndCenterElement(HelpUtils.wrapWithHelpButton(this.withEntry("item_extension/burn_time"),
+						L10N.label("elementgui.item_extension.burn_time")), PanelUtils.centerInPanel(fuelPower)))));
+
+		fuelPanel.setBorder(BorderFactory.createTitledBorder(
+				BorderFactory.createLineBorder((Color) UIManager.get("MCreatorLAF.BRIGHT_COLOR"), 1),
+				L10N.t("elementgui.item_extension.fuel_properties"), TitledBorder.LEADING,
+				TitledBorder.DEFAULT_POSITION, getFont(), (Color) UIManager.get("MCreatorLAF.BRIGHT_COLOR")));
+
+		// Composter

```

Comment:
Composter is a block, compostability is an item property.
```suggestion
		// Compostability
```

Reply:
So what about these 3 suggestions?